enhancements/network/ovs-dynamic-cpu-pinning.md

```diff
@@ -0,0 +1,203 @@
+---
+title: ovs-dynamic-cpu-pinning
+authors:
+  - "@zeeke"
+reviewers:
+  - "@trozet"
+  - "@MarSik"
+  - "@ricky-rav"
+  - "@cgoncalves"
+approvers:
+  - "@trozet"
+api-approvers:
+  - "None"
+creation-date: 2023-03-31
+last-updated: 2023-03-31
+tracking-link:
+  - https://issues.redhat.com/browse/CNF-5977
+---
+
+# OVS Dynamic CPU Pinning
+
+## Summary
+
+Allow OVS daemons to run on reserved and non-reserved CPUs if the network workload requires
+cycles and non-reserved CPU are idle.
+
+## Motivation
+
+OVS runs on reserved CPUs when an OCP node is configured with [cpuManagerPolicy](https://docs.openshift.com/container-platform/4.12/scalability_and_performance/using-cpu-manager.html) set to `static`, and the kubelet is configured with [`reservedSystemCpus`](https://kubernetes.io/docs/tasks/administer-cluster/reserve-compute-resources/#explicitly-reserved-cpu-list).
+
+Typically the users want to minimize the amount of reserved CPUs and leave 
+as many available CPUs as possible for business-related workloads.
+
+We have seen in some customer cases that it is possible to disrupt both the node
+and the cluster when the networking load rises, because OVS is CPU bound by the 
+limited number of system-reserved cores, and networking performance suffers.
+
+### User Stories
+
+> As an Openshift cluster administrator
+> I want to maximize my network dataplane bandwidth to all available cores in a dynamically reserved CPU environment
+> so that the cluster can endure a high network workload.
+
+### Goals
+
+Every cluster node that is configured with static CPU manager policy and
+a low number of `reservedSystemCpus` does not experience any excessive latency 
+increase while dealing with high network loads.
+
+### Non-Goals
+
+The logic used by Node Tuning Operator to decide if and when to enable the feature
+are out of this enhancement's scope.
+
+## Proposal
+
+As of today, a set of CPUs can be reserved for system workload (i.e. other than Kubernetes workload)
+by specifying the `--reserved-cpus` kubelet parameter (or `reservedSystemCpus` via configuration file).
+
+The main idea to resolve the issue described above is to allow OVS 
+to spread out to other cpus outside the reserved cpu pool. However 
+a care must be taken to not interfere with cpus that are assigned 
+to latency sensitive workloads.
+
+### Workflow Description
```

Comment:
This will not work for a workload partitioned system.

Reply:
Why not? The ovnkube-node pod will be assigned to reserved by WP and it will set the same affinity to OVS. We can disable the feature and get the same outcome too.

The question is, do we want to allow OVS to use more cpus in workload partitioned system?

Reply:
Yep, that I agree with indeed.

Reply:
@browsell @MarSik, do you think there is something to add to this enhancement about it? If so, please provide some links to workload partition system doc, so I can better understand the point